dask/tests/test_sizeof.py

```diff
@@ -43,6 +43,14 @@ def test_pandas():
     assert isinstance(sizeof(df.index), int)
 
 
+def test_pandas_multiindex():
+    pd = pytest.importorskip("pandas")
+    index = pd.MultiIndex.from_product([range(5), ["a", "b", "c", "d", "e"]])
+
+    assert 0.5 * sys.getsizeof(index) < sizeof(index) < 2 * sys.getsizeof(index)
```

Comment:
this test fails for current master with `AssertionError: assert 5222 < (2 * 796)`

Reply:
Do you have a sense for what is going on here?

Reply:
More broadly, which of the following are wrong?

1.  sys.getsizeof?
2.  dask.sizeof.sizeof
3.  Our assertion that sizeof should be close to sys.getsizeof?

Reply:
Sorry, what I meant was: for this new test, the seemingly arbitrary (0.5, 2)x range was chosen so that the old incorrect method would fail and the new one (which is still not exactly but closeish) would pass.

sys.getsizeof and .memory_usage agree and both seem right; the current Dask implemention is wrong for the reasons described in the linked issue, which this PR attempts to address

Reply:
Oh I hadn't noticed the failure on Travis; maybe a difference in behavior between Python or pandas versions? I'll look into it.